src/main/java/cz/sparko/boxitory/service/FilesystemBoxRepository.java

```diff
@@ -75,16 +73,41 @@ private boolean validateFilename(File boxFile) {
         return true;
     }
 
-    private BoxVersion parseBoxFile(File file) {
+    private String getBoxVersionFromFileName(File file) {
         String filename = file.getName();
         List<String> parsedFilename = Arrays.asList(filename.split("_"));
+        return parsedFilename.get(1);
+    }
+
+    private List<BoxVersion> createBoxVersionsFromGroupedFiles(Map<String, List<File>> groupedFiles) {
+        List<BoxVersion> boxVersions = new ArrayList<>();
+        groupedFiles
+                .entrySet()
+                .forEach(
```

Comment:
this can be done by single forEach. entrySet is needless here.
`groupedFiles.forEach((key, value) -> ...add...(key, value))`

imho all streams in this whole class can be done by one stream somehow. but this is sufficient for now... and maybe more clear for reader.